app/upgrades/v17/upgrades.go

```diff
@@ -124,12 +89,225 @@ func CreateUpgradeHandler(
+	commPoolBalancePre := sdk.NewCoins(commPoolBalanceBaseAssetPre, commPoolBalanceQuoteAssetPre)
+
+	// Swap 0.1 OSMO for baseAsset from the community pool.
+	respectiveBaseAssetInt, err := keepers.GAMMKeeper.SwapExactAmountIn(ctx, communityPoolAddress, linkedClassicPool, osmoIn, baseAsset, sdk.ZeroInt(), linkedClassicPool.GetSpreadFactor(ctx))
+	if err != nil {
+		return "", 0, err
+	}
+	ctx.Logger().Info(fmt.Sprintf("Swapped %s for %s%s from the community pool", osmoIn.String(), respectiveBaseAssetInt.String(), baseAsset))
+
+	respectiveBaseAsset := sdk.NewCoin(baseAsset, respectiveBaseAssetInt)
+
+	// Create a full range position via the community pool with the funds we calculated above.
+	fullRangeCoins := sdk.NewCoins(respectiveBaseAsset, osmoIn)
+	_, _, _, _, err = keepers.ConcentratedLiquidityKeeper.CreateFullRangePosition(ctx, clPoolId, communityPoolAddress, fullRangeCoins)
+	if err != nil {
+		return "", 0, err
+	}
+
+	// Get community pool balance after swap and position creation
+	commPoolBalanceBaseAssetPost := keepers.BankKeeper.GetBalance(ctx, communityPoolAddress, baseAsset)
+	commPoolBalanceQuoteAssetPost := keepers.BankKeeper.GetBalance(ctx, communityPoolAddress, QuoteAsset)
+	commPoolBalancePost := sdk.NewCoins(commPoolBalanceBaseAssetPost, commPoolBalanceQuoteAssetPost)
+
+	// While we can be fairly certain the diff between these two is 0.2 OSMO, if for whatever reason
+	// some baseAsset dust remains in the community pool and we don't account for it, when updating the
+	// fee pool balance later, we will be off by that amount and will cause a panic.
+	coinsUsed := commPoolBalancePre.Sub(commPoolBalancePost)
+
+	// Track the coins used to create the full range position (we manually update the fee pool later all at once).
+	*fullRangeCoinsUsed = fullRangeCoinsUsed.Add(coinsUsed...)
+
+	return clPoolDenom, clPoolId, nil
+}
+
+// authorizeSuperfluidIfEnabled authorizes superfluid for a CL pool if the balancer pool is superfluid enabled.
+func authorizeSuperfluidIfEnabled(ctx sdk.Context, keepers *keepers.AppKeepers, gammPoolId uint64, clPoolDenom string) (err error) {
+	// If pair was previously superfluid enabled, add the cl pool's full range denom as an authorized superfluid asset.
+	poolShareDenom := fmt.Sprintf("gamm/pool/%d", gammPoolId)
+	_, err = keepers.SuperfluidKeeper.GetSuperfluidAsset(ctx, poolShareDenom)
+	if err == nil {
+		ctx.Logger().Info(fmt.Sprintf("%s is superfluid enabled, enabling %s as a superfluid asset", poolShareDenom, clPoolDenom))
+		superfluidAsset := superfluidtypes.SuperfluidAsset{
+			Denom:     clPoolDenom,
+			AssetType: superfluidtypes.SuperfluidAssetTypeConcentratedShare,
+		}
+		err = keepers.SuperfluidKeeper.AddNewSuperfluidAsset(ctx, superfluidAsset)
+		if err != nil {
+			return err
+		}
+	}
+	ctx.Logger().Info(fmt.Sprintf("%s is not superfluid enabled, not enabling %s as a superfluid asset", poolShareDenom, clPoolDenom))
+	return nil
```

Comment:
Confirming - are we confident that error always implies superfluid being disabled?

Reply:
Confirming if key in store is not found, we return an error: https://github.com/osmosis-labs/osmosis/blob/59c5a8c077aef6bb0cbe739e90f7705ad33d66d5/x/superfluid/keeper/superfluid_asset_store.go#L41-L43